runelite-client/src/main/java/net/runelite/client/plugins/npcunaggroarea/NpcAggroAreaPlugin.java

```diff
@@ -0,0 +1,579 @@
+		coords[1] = lp.getY() - Perspective.LOCAL_TILE_SIZE / 2;
+	}
+
+	private void reevaluateActive()
+	{
+		updateTimer();
+		calculateLinesToDisplay();
+	}
+
+	private void calculateLinesToDisplay()
+	{
+		if (active && config.showAreaLines())
+		{
+			Rectangle sceneRect = new Rectangle(
+				client.getBaseX() + 1, client.getBaseY() + 1,
+				Constants.SCENE_SIZE - 2, Constants.SCENE_SIZE - 2);
+
+			for (int i = 0; i < linesToDisplay.length; i++)
+			{
+				currentPlane = i;
+
+				GeneralPath lines = new GeneralPath(generateSafeArea());
+				lines = Geometry.clipPath(lines, sceneRect);
+				lines = Geometry.unitifyPath(lines, 1);
+				if (config.collisionDetection())
+				{
+					lines = Geometry.filterPath(lines, this::collisionFilter);
+				}
+				lines = Geometry.transformPath(lines, this::transformWorldToLocal);
+				linesToDisplay[i] = lines;
+			}
+		}
+		else
+		{
+			for (int i = 0; i < linesToDisplay.length; i++)
+			{
+				linesToDisplay[i] = null;
+			}
+		}
+	}
+
+	private void removeTimer()
+	{
+		if (currentTimer != null)
+		{
+			infoBoxManager.removeInfoBox(currentTimer);
+			currentTimer = null;
+		}
+	}
+
+	private void createTimer(Instant startTime)
+	{
+		removeTimer();
+
+		if (config.showTimer() && active)
+		{
+			BufferedImage image = itemManager.getImage(ItemID.ENSOULED_DEMON_HEAD);
+			currentTimer = new Timer(startTime, AGGRESSIVE_TIME_SECONDS, ChronoUnit.SECONDS, image, this);
+			currentTimer.setTooltip("Time until NPCs become unaggressive");
+			infoBoxManager.addInfoBox(currentTimer);
+		}
+	}
+
+	private void updateTimer()
+	{
+		if (endTime == null)
+		{
+			removeTimer();
+		}
+		else
+		{
+			Instant startTime = endTime.minus(Duration.ofSeconds(AGGRESSIVE_TIME_SECONDS));
+			createTimer(startTime);
+		}
+	}
+
+	private void resetTimer()
+	{
+		endTime = Instant.now().plusSeconds(AGGRESSIVE_TIME_SECONDS);
+		updateTimer();
+	}
+
+	private boolean isNpcMatch(NPC npc)
+	{
+		NPCComposition composition = npc.getTransformedComposition();
+		if (composition == null)
+		{
+			return false;
+		}
+
+		// Most NPCs stop aggroing when the player has more than double
+		// its combat level.
+		int playerLvl = client.getLocalPlayer().getCombatLevel();
+		int npcLvl = composition.getCombatLevel();
+		String npcName = composition.getName().toLowerCase();
+		if (npcLvl > 0 && playerLvl > npcLvl * 2)
+		{
+			return false;
+		}
+
+		for (String pattern : npcNamePatterns)
+		{
+			if (WildcardMatcher.matches(pattern, npcName))
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private void recheckActive()
+	{
+		active = config.alwaysActive();
+
+		if (!active)
+		{
+			for (NPC npc : client.getNpcs())
+			{
+				if (isNpcMatch(npc))
+				{
+					active = true;
+					break;
+				}
+			}
+		}
+
+		reevaluateActive();
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		if (config.alwaysActive())
+		{
+			return;
+		}
+
+		if (!active && isNpcMatch(event.getNpc()))
```

Comment:
you should make this common method that accepts var amount of NPCs and do the reevaluate active and active true there (check ClueScrollPlugin what does basically the same thing)